src/main/java/com/shoppinglist/shoppinglist/service/ShoppingListService.java

```diff
@@ -28,11 +32,22 @@ public List<ListItem> fetchAll() {
 
     @Override
     public ListItem fetchById(int id) throws NoSuchElementException {
-        return shoppingListDAO.fetch(id);
+        try {
+            return shoppingListDAO.fetch(id);
+        } catch (NoSuchElementException e) {
+            logger.error("Item not found with ID: {}", id);
+            throw e;
+        }
     }
 
     @Override
     public void delete(int id) throws NoSuchElementException {
-        shoppingListDAO.delete(id);
+        try {
+            shoppingListDAO.delete(id);
+            logger.info("Deleted item with ID: {}", id);
+        } catch (NoSuchElementException e) {
+            logger.error("Item not found with ID: {}", id);
+            throw e;
+        }
     }

```

Comment:
An exception handler block is used to catch the NoSuchElementException exception, and then an error log is recorded showing the ID of the unfound item.It then proceeds to rethrow the exception so that the upper level caller can also see the exception.